docs/contribute/plugin/plugin_requirement.md

```diff
@@ -0,0 +1,182 @@
+---
+sidebar_position: 2
+---
+
+# Requirements for Developing WasmEdge Plugin
+
+By developing a plugin, one can extend the functionality of WasmEdge and customize it to suit the specific needs. There are some steps needs to be followed for developing WasmEdge Plugin: 
+
+## Choose a programming language
+
+The first step in developing a WasmEdge plugin is to choose the programming language one wants to use. Currently, C, C++, and Rust are supported by the WasmEdge runtime.
+
+- **C:** For developing the WasmEdge plug-in in C API, please [install WasmEdge](/develop/build-and-run/install.md) first.
+  
+- **C++:** For developing the WasmEdge plug-in in internal C++, developers should [build WasmEdge from source](../source/build_from_src.md) before starting. On the other hand, to build a plugin using internal C++, they should build WasmEdge from source.
+
+- **Rust:** Rust is another option that has gained popularity due to its safety features and performance. It is a systems programming language designed for safe, concurrent, and fast execution.
+
+The choice of programming language will depend on the developer's familiarity and skill with the language. However, C and C++ are popular choices for developers who want to create high-performance plugins with low-level control over system resources. Rust is another popular choice for its safety features and performance.
+
+> We recommend developers to develop plug-ins in WasmEdge [C API](develop_plugin_c.md).
+
+## Set up the development environment
+
+To set up your development environment for WasmEdge plugin development, follow these steps:
+
+- **Install a WasmEdge runtime**: Choose a runtime environment that can execute WebAssembly modules. Options include WasmEdge, WASM3, or WAVM, depending on your requirements.
```

Comment:
The WasmEdge Plugin only works for the WasmEdge runtime. You only need to take WasmEdge as an example.